include/ddc/kernels/splines/bsplines_non_uniform.hpp

```diff
@@ -148,28 +137,26 @@ class NonUniformBSplines
 
         KOKKOS_INLINE_FUNCTION ddc::Coordinate<Tag> get_knot(int knot_idx) const noexcept
         {
-            return coord_from_knot(
-                    ddc::coordinate(ddc::DiscreteElement<mesh_type>(knot_idx + degree())));
+            return ddc::coordinate(ddc::DiscreteElement<mesh_type>(knot_idx + degree()));
         }
 
         KOKKOS_INLINE_FUNCTION ddc::Coordinate<Tag> get_first_support_knot(
                 discrete_element_type const& ix) const
         {
-            return coord_from_knot(ddc::coordinate(ddc::DiscreteElement<mesh_type>(ix.uid())));
+            return ddc::coordinate(ddc::DiscreteElement<mesh_type>(ix.uid()));
         }
```

Comment:
Are these functions still necessary? Do we not have a non-uniform point sampling for the knots?

Reply:
Same answer